File: ios/Sources/CapLiveActivitiesPlugin/CapLiveActivities.swift
import Foundation

@objc public class CapLiveActivities: NSObject {
    public static var liveActivityDelegate: LiveActivityDelegate?

    @objc public func echo(_ value: String) -> String {
        print(value)
        return value
    }

    public func startLiveActivity(testString: String? = nil) -> Bool {
        if #available(iOS 16.1, *) {
            if let test = testString {
                print("Test string received:", test)
            }
            // CapLiveActivities.liveActivityDelegate?.monitorPushToStartToken()
            // CapLiveActivities.liveActivityDelegate?.monitorNewActivities()
            CapLiveActivities.liveActivityDelegate?.initialize()
            return true
        } else {
            print("Live Activities require iOS 16.1 or later")
            return false
        }
    }
}

File: android/src/main/java/com/mycompany/plugins/example/CapLiveActivities.java
package com.mycompany.plugins.example;

import android.util.Log;

public class CapLiveActivities {

    public String echo(String value) {
        Log.i("Echo", value);
        return value;
    }

    public boolean startLiveActivity(String testString) {
        Log.w("CapLiveActivities", "Live Activities are not supported on Android");
        if (testString != null) {
            Log.i("CapLiveActivities", "Test string: " + testString);
        }
        return false;
    }
}
